src/mds/nameserver2/namespace_storage.cpp

```diff
@@ -40,8 +40,12 @@ std::ostream &operator<<(std::ostream &os, StoreStatus &s) {
 }
 
 NameServerStorageImp::NameServerStorageImp(
-    std::shared_ptr<KVStorageClient> client, std::shared_ptr<Cache> cache)
-    : cache_(cache), client_(client), discardMetric_() {}
+    std::shared_ptr<KVStorageClient> client, std::shared_ptr<Cache> cache, bool is_kvstorage )
+    : cache_(cache), client_(client), discardMetric_(),is_kvstorage_(is_kvstorage) {}
+
+NameServerStorageImp::NameServerStorageImp(
+    std::shared_ptr<MysqlClientImp> mysqlclient, std::shared_ptr<Cache> cache, bool is_kvstorage )
```

Comment:
It is not recommended to use `is_kvstorage` to differentiate storage types. If more types of storage are supported in the future, there will be a lot of if-else statements in the code. Instead, the "KVStorageClient" can be renamed as "StorageClient" and used as the interface definition for all clients.

Based on the implementation below, the basic interfaces such as `Put, Delete, and Get ` are called uniformly, which is not a problem. However, the code related to transactions has review suggestions in the corresponding places.